server/domain/repository/auth_repository.go

```diff
@@ -0,0 +1,11 @@
+package repository
+
+import (
+	"context"
+)
+
+type IAuthRepository interface {
+	GetGithubOAuthURL(ctx context.Context, state string) string
+	GetGithubOAuthToken(ctx context.Context, code string) (string, error)
+	GetGithubUser(ctx context.Context, token string) (map[string]interface{}, error)
```

Comment:
```suggestion
	GetOAuthURL(ctx context.Context, state string) string
	GetOAuthToken(ctx context.Context, code string) (string, error)
	GetOAuthUser(ctx context.Context, token string) (map[string]interface{}, error)
```

関数名からGithubけしたいんだけどどう？
この方がたとえば後でTwitterとかの認証が増えた時にも対応できると思うんだけど

Reply:
いまいちイメージがわかないんだけど、Twitterとかの認証増えた場合って同じrepositoryで実装するの？？
GithubとTwitterで分けるんじゃなくて？？

Reply:
うん、俺のイメージ的にはOAuthProviderっていう外部レイヤーのイメージだったけどな
ドメイン内のリポジトリに外部の名前入るの気持ち悪くない？

Reply:
あーなんとなくだけどイメージ沸いた
あと、ドメイン内に外部の名前があるのは確かにきもいな

Reply:
方針は自分が提案しているものでなくてもいいけど、とりあえずGithubっていう名前を含めたくないなっていう
それでいうと`map[string]interface{}`もgithubフィールドのものになっているから`OAuthUserResponse`みたいなstructで共通化してから返した方がいいかも

Reply:
んー、一瞬気持ち悪いかなと思ったけどそうでもなくなってきた
```
domain/
|-- entities/
|-- repositories/
    |-- twitter_repository_interface.go
    |-- google_repository_interface.go
    |-- github_repository_interface.go
```
俺のイメージ認証増えた時はこんな感じにすると思ってた